### PR TITLE
Optimize analysis calculation

### PR DESCRIPTION
**Pull request by fpetkovski**
Aggregating analyzed results is slow for complex expressions because we aggregate samples over and over again.

This commit speeds that up by memoizing aggregates.

### CODE REVIEW

File: logicalplan/distribute.go [hunk 1]
```diff
@@ -198,9 +198,9 @@ func (m DistributedExecutionOptimizer) Optimize(plan Node, opts *query.Options)
 				return true
 			}
 
-			sum := *(*current).(*Aggregation)
+			sum := *(*current).Clone().(*Aggregation)
 			sum.Op = parser.SUM
-			count := *(*current).(*Aggregation)
+			count := *(*current).Clone().(*Aggregation)
```
Comment (pedro-stanaka) [inline logicalplan/distribute.go:203]: Is this related?

Comment (fpetkovski) [reply logicalplan/distribute.go]: Good catch, reverted.

**Review by MichaHoffmann — APPROVED** [review]

**Review by saswatamcode — APPROVED** [review]
Thanks!
